### ISSUE TITLE
Input events for most keys have scancode 0 in HTML5 export

### ISSUE THREAD

**Issue by eska014** [godotengine/godot#2761]
The affected keys will fire an InputEventKey event, but scancode is given as 0.
The same keys will also not be recognized by the Input singleton, nor will actions be fired by them.
This occurs with most keys, exceptions are:
 -  Space, Insert
 -  Up, Down, Left, Right
 -  Home, End, PageUp/Down
 -  F1-F12

Previously reported in #2550

**Comment by akien-mga** [comment]
I confirm that, at least it was the case a month ago when I tried, and I don't think it has change since.

**Comment by reduz** [comment]
emscripten glut does not support unicode.. i think.. or i couldn't get it to work. Can ask Azakai i guess

**Comment by FEDE0D** [comment]
I don't get it. Was this fixed?
I'm using the 2.0.1 build and this is still happening.

**Comment by eska014** [comment]
It's not fixed. Emscripten's [`html5.h`](https://kripken.github.io/emscripten-site/docs/api_reference/html5.h.html) offers callbacks for JavaScript input events, but those things are a [compatibility mess](https://kripken.github.io/emscripten-site/docs/api_reference/html5.h.html#c.EmscriptenKeyboardEvent)

**Comment by FEDE0D** [comment]
Is there a workaround for that?

**Comment by eska014** [comment]
I have scancodes working after lots of tedious testing and typing: https://github.com/eska014/godot/commit/244476d65d28bcae80a01803506649b066b1eac1
But finding unicode values is difficult.

I set up a tool to test JavaScript's `KeyboardEvent` properties on different browser&OS combinations at https://godot.eska.me/web-key-test/.
Getting unicode values is easy in `keypress` events, but linking those to `keydown` (which is fired just before keypress) and `keyup` event is hard, if not impossible.
The `key` and sometimes `char` properties could work in `keydown` and `keyup`, but are sparsely supported.

Best attempt to find unicode values is probably something like this:

 1. See if `key` is a string with length of 1, if so, use it
 2. See if `char` is a string with length of 1, if so, use it (probably just IE)
 3. Tediously try to figure out the character from the remaining properties (this method has been used in web export so far, with bad results)
 4. Give up

This is what the linked commit is doing as well, except for stage 3

 - Internet Explorer 11 has `char`, which seems to work somewhat (though `char` was dropped from the standard)
 - Firefox just gives `MozPrintableKey` for printable characters in `key` on Windows, but gives usable values on X11

 - Chromium 51 doesn't offer anything salvageable at all, neither `key` nor `char`. May have changes eventually:
  https://bugs.chromium.org/p/chromium/issues/detail?id=227231
  https://bugs.chromium.org/p/chromium/issues/detail?id=542598

**tldr: Finding unicode values for key events is difficult because browsers lag behind standards, we'll have to either say "make sure your browser supports `KeyboardEvent.key`" or hack it**

**Comment by eska014** [comment]
Running Chromium with `--enable-experimental-web-platform-features` makes the `key` property available. In Firefox Developer Edition, Firefox also supports `key` on Windows. The question is when they'll finally make it to stable

**Comment by eska014** [comment]
Got unicode values working in browsers without `key` and `char` by deferring keydown to keypress events.

Release events only have `unicode` value set in browsers supporting the new properties, but other OS don't set unicode at all in release event anyway
